Tests/CacheAdvanceTests/CacheAdvanceTests.swift

```diff
@@ -652,5 +678,38 @@ final class CacheAdvanceTests: XCTestCase {
             attributes: nil)
     }
 
+    /// Creates a cache file with incorrect offset in header.
+    /// - Parameter messageCount: The number of messages to write to the cache file
+    /// Returns the header handle of the cache.
+    private func prepareIncorrectOffsetInHeaderAndWriteMessage(messageCount: Int) throws -> CacheHeaderHandle {
+        let randomHighValue: UInt64 = 100_000
```

Comment:
This high value was truly chosen at random in the old test which contained this code – it did not matter what the value was because _no messages were written to the file in this test_.

Now that you've changed this code to actually write messages, we should make sure that the size of the cache is appropriate for the parameters passed into this method.

I think this `randomHighValue` should be _added_ to byte size that is correct for the number of messages we're going to write. Maybe instead of a `messageCount` parameter, we could pass in the messages we want the file to contain as an array. Then we could use the `requiredByteCount` to find the appropriate size for those messages, and then `maximumBytes` would be set to that required byte count _plus_ `randomHighValue`.

As written, this method could unintentionally `throw` `CacheAdvanceError.messageLargerThanRemainingCacheSize` if `messageCount` is high. I'd like to avoid that.

Reply:
good idea to pass the messages as parameters, let me do that.